Sources/DataSources/ASTableNode+Rx/ASTableSectionedDataSource.swift

```diff
@@ -35,7 +35,8 @@ open class ASTableSectionedDataSource<S: SectionModelType>: NSObject, ASTableDat
     }
 
     fileprivate static func configureCellBlockNotSet(dataSource: ASTableSectionedDataSource<S>, node: ASTableNode, indexPath: IndexPath, model: I) -> ASCellNodeBlock {
-        return { dataSource.tableNode(node, nodeForRowAt: indexPath) }
+        // Users expect taleNode(_:nodeForRowAt:) will be executed in main thread according to api doc http://texturegroup.org/appledoc/Protocols/ASTableDataSource.html#//api/name/tableNode:nodeForRowAtIndexPath: .
+        return { DispatchQueue.main.sync(execute: { dataSource.tableNode(node, nodeForRowAt: indexPath) }) }
```

Comment:
IMHO, Binder using in `RxASTableNodeReloadDataSource` already guarantee that datasource is executed on Main Thread.

Could you help to verify that by using another DispatchQueue in Example app.

Reply:
Sorry for delay.

`configureCellBlockNotSet` may be executed in `tableNode(_ tableNode:nodeBlockForRowA:) -> ASCellNodeBlock` and this delegate is **NOT** called in main thread but called in background thread.

Following sentences are from [Texture API Doc](http://texturegroup.org/appledocs.html).

```
– tableNode:nodeBlockForRowAtIndexPath:

Asks the data source for a block to create a node to represent the row at the given index path.
The block will be run by the table node concurrently in the background before the row is inserted into the table view.
```

Reply:
@fumito-ito @dangthaison91 
I don't think this is best solution.
```
 tableNode:nodeForRowAt:
 tableNode:nodeBlockForRowAt:
```  
will always execute in main thread.
because ASDK will check execution thread in ASDataController by `ASDisplayNodeAssertMainThread()`.
but ASNodeBlock() will be execute concurrently in background .
So, see the Rx logic, if we not set configureCellBlock, it will fall back by embed `tableNode:nodeForRowAt:`  in `configureCellBlockNotSet`, It will cause `tableNode:nodeForRowAt:` execute in background thread.
Actually, without RxASDataSOurce, if you want init node at main thread, you should just implement `tableNode:nodeForRowAt:` and ignore `tableNode:nodeBlockForRowAt:`, but both of them implement in Rx, I guess this is why we use `configureCellBlockNotSet` function.
in the end, this is my solution: 
```
 let cellNode = dataSource.tableNode(node, nodeForRowAt: indexPath)
 return { cellNode }
```

Reply:
Thank you for your comment @PatrickChow .

Your solution looks same as `return { dataSource.tableNode(node, nodeForRowAt: indexPath) }`. Do you mean this change is not needed ?

Reply:
Oh I found the difference. According to your solution, `let cellNode = dataSource.tableNode(node, nodeForRowAt: indexPath)` is called in man thread but `{ cellNode }` will be executed in background.

Reply:
@fumito-ito Yes, it is